util/azure/utils.go

```diff
@@ -0,0 +1,341 @@
+	} else {
+		cfg, err = rest.InClusterConfig()
+	}
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Error building kubeconfig")
+	}
+
+	kubeclientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error building kubernetes clientset")
+	}
+
+	apiresourcesList, err := kubeclientset.Discovery().ServerPreferredResources()
+	if err != nil {
+		return nil, err
+	}
+
+	printApiresourcesList, _ := json.Marshal(apiresourcesList)
+
+	klog.V(5).Infof("List of ApiResources fetched from apiserver: %s", string(printApiresourcesList))
+
+	return apiresourcesList, nil
+}
+
+func fetchDataActionsList(environment string, clusterType string, loginURL string, tenantID string, clientID string, clientSecret string) ([]Operation, error) {
+	env := azure.PublicCloud
+	var err error
+	if environment != "" {
+		env, err = azure.EnvironmentFromName(environment)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to parse environment for Azure.")
+		}
+	}
+
+	endpoint := ""
+
+	switch clusterType {
+	case ConnectedClusters:
+		endpoint = fmt.Sprintf(OperationsEndpointFormatARC, env.ResourceManagerEndpoint)
+	case ManagedClusters:
+		endpoint = fmt.Sprintf(OperationsEndpointFormatAKS, env.ResourceManagerEndpoint)
+	case Fleets:
+		endpoint = fmt.Sprintf(OperationsEndpointFormatAKS, env.ResourceManagerEndpoint)
+	default:
+		return nil, errors.Errorf("Failed to create endpoint for Get Operations call. Cluster type %s is not supported.", clusterType)
+	}
+
+	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create request for Get Operations call.")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", fmt.Sprintf("guard-%s-%s-%s", v.Version.Platform, v.Version.GoVersion, v.Version.Version))
+
+	var token string
+	if clusterType == ConnectedClusters {
+		tokenProvider := graph.NewClientCredentialTokenProvider(clientID, clientSecret,
+			fmt.Sprintf("%s%s/oauth2/v2.0/token", env.ActiveDirectoryEndpoint, tenantID),
+			fmt.Sprintf("%s/.default", env.ResourceManagerEndpoint))
+
+		authResp, erro := tokenProvider.Acquire("")
+		if erro != nil {
+			return nil, errors.Wrap(erro, "Error getting authorization headers for Get Operations call.")
+		}
+
+		token = authResp.Token
+	} else { // AKS and Fleet
+		tokenProvider := graph.NewAKSTokenProvider(loginURL, tenantID)
+
+		authResp, err := tokenProvider.Acquire("")
+		if err != nil {
+			return nil, errors.Wrap(err, "Error getting authorization headers for Get Operations call.")
+		}
+
+		token = authResp.Token
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+
+	client := httpclient.DefaultHTTPClient
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to send request for Get Operations call.")
+	}
+	defer resp.Body.Close()
+
+	data, err := ioutil.ReadAll(resp.Body)
```

Comment:
io/ioutil is deprecated. prefer to using os.ReadAll
https://pkg.go.dev/io/ioutil#pkg-overview

Reply:
done